L.Control.Layers.Tree.d.ts

```diff
@@ -0,0 +1,74 @@
+declare module "leaflet.control.layers.tree" {
+  module "leaflet" {
+    interface ILayerTreeChildren {
+      label: string;
+      layer: L.Layer;
+      selectAllCheckbox?: boolean;
+      children?: ILayerTreeChildren[];
+    }
+
+    interface ILayerTree {
+      label: string;
+      children?: ILayerTreeChildren[];
+    }
+
+    interface ILayerTreeOptions {
+      /**
+       * Symbol displayed on a closed node.
+       * @default '+'
+       */
+      closedSymbol: string;
+      /**
+       * Symbol displayed on an open node.
+       * @default '−' (&minus;)
+       */
+      openSymbol: string;
+      /**
+       * Symbol between the closed or opened symbol, and the text. Default .
+       * @default ' ' (a normal space)
+       */
+      spaceSymbol: boolean;
+      /**
+       * Flag to indicate if the selector (+ or −) is after the text.
+       * @default 'false'
+       */
+      selectorBack: boolean;
+      /**
+       * Flag to replace the toggle image (box with the layers image) with the 'name' of the selected base layer.
+       * If the name field is not present in the tree for this layer, label is used.
+       * See that you can show a different name when control is collapsed than the one that appears in the tree
+       * when it is expanded. Your node in the tree can be
+       * { label: 'OSM', name: 'OpenStreetMap', layer: layer }.
+       * @default 'false'.
+       */
+      namedToggle: boolean;
+      /**
+       * Text for an entry in control that collapses the tree (baselayers or overlays).
+       * If empty, no entry is created.
+       * @default ''.
+       */
+      collapseAll: string;
+      /**
+       * Text for an entry in control that expands the tree. If empty, no entry is created.
+       * @default ''.
+       */
+      expandAll: string;
+      /**
+       * Controls if a label or only the checkbox/radiobutton can toggle layers.
+       * If set to both, overlay or base those labels can be clicked on to toggle the layer.
+       * @default 'both'.
+       */
+      labelIsSelector: string;
+    }
+
+    namespace control {
+      namespace layers {
+        function tree(
+          baseTree: ILayerTree,
+          overlayTree: ILayerTree,
+          options?: Partial<ILayerTreeOptions>
+        ): L.Control;
+      }
+    }
+  }
+}
```

Comment:
Sorry, I am not an expert in TypeScript. So I asked a friend of mine.
They suggested this full definition.

(I do not know a better way to give you the full file)
```suggestion
import type L from 'leaflet';

export namespace Control {
  export namespace Layers {
    interface LayerTreeChildren {
      label: string;
      layer: L.Layer;
      selectAllCheckbox?: boolean;
      children?: LayerTreeChildren[];
    }

    interface LayerTree {
      label: string;
      children?: LayerTreeChildren[];
    }

    interface LayerTreeOptions extends L.Control.LayersOptions {
      /**
      * Symbol displayed on a closed node.
      * @default '+'
      */
      closedSymbol?: string;

      /**
      * Symbol displayed on an open node.
      * @default '−' (&minus;)
      */
      openedSymbol?: string;

      /**
      * Symbol between the closed or opened symbol, and the text. Default .
      * @default ' ' (a normal space)
      */
      spaceSymbol?: boolean;

      /**
      * Flag to indicate if the selector (+ or −) is after the text.
      * @default 'false'
      */
      selectorBack?: boolean;

      /**
      * Flag to replace the toggle image (box with the layers image) with the 'name' of the selected base layer.
      * If the name field is not present in the tree for this layer, label is used.
      * See that you can show a different name when control is collapsed than the one that appears in the tree
      * when it is expanded. Your node in the tree can be
      * { label: 'OSM', name: 'OpenStreetMap', layer: layer }.
      * @default 'false'.
      */
      namedToggle?: boolean;

      /**
      * Text for an entry in control that collapses the tree (baselayers or overlays).
      * If empty, no entry is created.
      * @default ''.
      */
      collapseAll?: string;

      /**
      * Text for an entry in control that expands the tree. If empty, no entry is created.
      * @default ''.
      */
      expandAll?: string;

      /**
      * Controls if a label or only the checkbox/radiobutton can toggle layers.
      * If set to both, overlay or base those labels can be clicked on to toggle the layer.
      * @default 'both'.
      */
      labelIsSelector?: 'both' | 'overlay' | 'base' | string;
    }

    class Tree extends L.Control.Layers {
      constructor(baseTree?: LayerTree, overlayTree?: LayerTree, options?: Partial<LayerTreeOptions>);
      addBaseLayer(layer: L.Layer, name: string): never;
      addOverlay(layer: L.Layer, name: string): never;
      removeLayer(layer: L.Layer): never;
      expandTree(isOverlay?: boolean): this;
      collapseTree(isOverlay?: boolean): this;
      expandSelected(isOverlay?: boolean): this;
      options: LayerTreeOptions;
    }
  }
}

export namespace control {
  export namespace layers {
    function tree(
      baseTree?: Control.Layers.LayerTree,
      overlayTree?: Control.Layers.LayerTree,
      options?: Partial<Control.Layers.LayerTreeOptions>
    ): Control.Layers.Tree;
  }
}
```

Reply:
@jjimenezshaw Sorry for the delay - missed your first revision. I've tried to copy paste it in our project (https://github.com/reiniernl/nederland-waterland), but alas, that did not work as expected, and I did get Typescript errors in my code. Did you do something special too?

Reply:
Hi @erikvullings . Could you check #36 ?
Pay attention to `files` in `test-types/tsconfig.json` https://github.com/jjimenezshaw/Leaflet.Control.Layers.Tree/pull/36/files#diff-d2828f04b5fc8ca0298f1358b0f50c12efcd37281d152f941b6da0a4697c3e57R11